src/main/java/com/gmail/inverseconduit/commands/sets/ChatCommands.java

```diff
@@ -2,47 +2,57 @@
 
 import java.util.logging.Logger;
 
+import com.gmail.inverseconduit.BotConfig;
 import com.gmail.inverseconduit.SESite;
 import com.gmail.inverseconduit.chat.ChatInterface;
 import com.gmail.inverseconduit.commands.CommandHandle;
 import com.gmail.inverseconduit.datatype.SeChatDescriptor;
 
 public final class ChatCommands {
 
-    public static CommandHandle unsummonCommand(ChatInterface chatInterface) {
-        return new CommandHandle.Builder("unsummon", message -> {
-            SeChatDescriptor descriptor = SeChatDescriptor.buildSeChatDescriptorFrom(message);
-            chatInterface.leaveChat(descriptor);
-            return "*~bye, bye*";
-        }).build();
-    }
+	public static CommandHandle unsummonCommand(ChatInterface chatInterface, BotConfig bc) {
+		return new CommandHandle.Builder("unsummon", message -> {
+			if (AdminCommands.isElevatedUser(message.getUserId(), bc)) {
+				SeChatDescriptor descriptor = SeChatDescriptor.buildSeChatDescriptorFrom(message);
+				chatInterface.leaveChat(descriptor);
+				return "*~bye, bye*";
+			} return "I am afraid, I cannot let you do that!!";
```

Comment:
yup. the privilege check should be baked into CommandHandle. We just need to keep track of whether a command is a privileged command and then check the admin status before making the message available